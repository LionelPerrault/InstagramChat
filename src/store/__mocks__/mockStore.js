import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

// mock-reducers
import User from './reducers/User'
import Notification from './reducers/Notification'
import Group from './reducers/Group'
import Follow from './reducers/Follow'
import Explore from './reducers/Explore'
import Post from './reducers/Post'
import Hashtag from './reducers/Hashtag'
import Setting from './reducers/Setting'
import Message from './reducers/Message'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore,
  Post,
  Hashtag,
  Setting,
  Message,
}

const middlewares = [thunk]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
