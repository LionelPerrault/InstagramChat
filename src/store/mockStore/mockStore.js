import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import promise from 'redux-promise-middleware'

// mock-reducers
import User from './mock-reducers/User'
import Notification from './mock-reducers/Notification'
import Group from './mock-reducers/Group'
import Follow from './mock-reducers/Follow'
import Explore from './mock-reducers/Explore'
import Post from './mock-reducers/Post'
import Hashtag from './mock-reducers/Hashtag'
import Setting from './mock-reducers/Setting'
import Message from './mock-reducers/Message'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore,
  Post,
  Hashtag,
  Setting,
  Message
}

const middlewares = [
  promise(), thunk
]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
