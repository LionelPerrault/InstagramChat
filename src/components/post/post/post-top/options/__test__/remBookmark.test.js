import React from 'react'
import MockDataElement from '../../../../../../utils/__mocks__/mock-dataElement'
import { create } from 'react-test-renderer'
import { Provider } from 'react-redux'
import mockStore from '../../../../../../store/__mocks__/mockStore'
import RemBookmarkAsAdmin, { PureRemBookmarkAsAdmin } from '../remBookmark'
import { shallow } from 'enzyme'

describe('RemBookmarkAsAdmin Component', () => {
  let dataElement

  beforeEach(() => (dataElement = MockDataElement()))

  afterEach(() => dataElement.remove())

  const props = {
    post_id: 11,
    user: 24,
    when: 'bookmarks',
  }

  it('should match snapshot', () => {
    dataElement.setAttribute('data-isadmin', 'true')
    const tree = create(
      <Provider store={mockStore}>
        <RemBookmarkAsAdmin {...props} />
      </Provider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should mock removeBookmark action when remove option is clicked', () => {
    dataElement.setAttribute('data-isadmin', 'true')
    const wrapper = shallow(
      <PureRemBookmarkAsAdmin {...props} dispatch={jest.fn()} />
    )
    wrapper.find('li > a').simulate('click', {
      preventDefault() {},
    })
  })
})
