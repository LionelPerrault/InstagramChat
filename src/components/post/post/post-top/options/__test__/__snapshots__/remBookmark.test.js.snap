// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemBookmarkAsAdmin Component should match snapshot 1`] = `
<li>
  <a
    href="#"
    onClick={[Function]}
  >
    Remove bookmark as admin
  </a>
</li>
`;
